pkg/operation/botanist/component/vpa/general.go

```diff
@@ -27,6 +30,31 @@ import (
 	"k8s.io/utils/pointer"
 )
 
+const (
+	metricsPortName = "metrics"
+)
+
+// getDefaultLivenessProbe returns a new instance of the default liveness probe configuration, used for VPA components
+func getDefaultLivenessProbe() *corev1.Probe {
+	// Return a copy as insurance against unintentional aliasing
```

Comment:
I'm not sure what that comment means – what is it that we're avoiding here?

Reply:
I'm avoiding what would happen with the alternative of reusing the same static object (vs. copy), and delegating the responsibility for not changing it to the caller:
a = getDefaultLivenessProbe()
b = getDefaultLivenessProbe()
a.ProbeHandler.Port = 80 // b.ProbeHandler.Port is unintentionally changed to 80

Reply:
I cannot follow what you mean. The case you described above does not seem to be right:
```golang
func main() {
	a := newDefaultLivenessProbe()
	b := newDefaultLivenessProbe()

	a.ProbeHandler.HTTPGet.Port = intstr.FromInt(80)
	fmt.Printf("%+v\n", a.ProbeHandler.HTTPGet.Port)
	fmt.Printf("%+v\n", b.ProbeHandler.HTTPGet.Port)
}
```

This outputs

```
{Type:0 IntVal:80 StrVal:}
{Type:1 IntVal:0 StrVal:metrics}
```

=> `b.ProbeHandler.HTTPGet.Port` does not get unintentionally changed

---

I guess you could remove the confusing comment.
```suggestion
```

Reply:
@andrerun the scenario you're describing would happen when the `newDefaultLivenessProbe()` implementation would look like this – in all other cases this should work as @ialidzhikov's example run showed.
```
var p = corev1.Probe{
	ProbeHandler: corev1.ProbeHandler{
		HTTPGet: &corev1.HTTPGetAction{
			Path:   "/health-check",
			Port:   intstr.FromString("metrics"),
			Scheme: corev1.URISchemeHTTP,
		},
	},
	InitialDelaySeconds: 120,
	PeriodSeconds:       60,
	TimeoutSeconds:      30,
	SuccessThreshold:    1,
	FailureThreshold:    3,
}

func newDefaultLivenessProbe() *corev1.Probe {
	return &p
}
```
So I guess let's remove the comment and keep the implementation, this is fine like it is.

Reply:
@voelzmo @ialidzhikov My goal with the comment was to highlight the intent of returning a fresh copy of the object upon each call, as opposed to reusing a singleton static instance. I felt I needed the comment, especially because I remembered seeing the other approach also used in Gardener source code in the past - large mutable objects stored in static variables. However, after prompted by your remarks, I did look more closely, and it turns out that the pattern is only used in tests, where [aliasing](https://en.wikipedia.org/wiki/Aliasing_(computing)) was not a risk.

Generally I prefer to err on the side of more code comments, to reduce the risk that I fall prey to the [curse of knowledge bias](https://en.wikipedia.org/wiki/Curse_of_knowledge). But, seeing that you both agree that the intent of that particular piece of code is immediately apparent, off with the comment.